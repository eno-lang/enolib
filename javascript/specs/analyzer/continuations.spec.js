const { analyze } = require('./analyze_helper.js');

const input = `
Field: Value
\\ Line continuation
| Newline continuation

Fieldset:
Empty = Value
\\ Line continuation
| Newline continuation

List:
- Value
\\ Line continuation
| Newline continuation

Empty field:
\\ Line continuation
| Newline continuation

Fieldset with empty entry:
Empty entry =
\\ Line continuation
| Newline continuation

List with empty item:
-
\\ Line continuation
| Newline continuation
`.trim();

describe('Continuation analysis', () => {
  it('performs as specified', () => {
    expect(analyze(input)).toMatchSnapshot();
  });
});
