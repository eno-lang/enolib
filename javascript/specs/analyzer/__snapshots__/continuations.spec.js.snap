// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continuation analysis performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [
      Object {
        "continuations": Array [
          Object {
            "line": 1,
            "ranges": Object {
              "line": Array [
                13,
                32,
              ],
              "spacedLineContinuationOperator": Array [
                13,
                14,
              ],
              "value": Array [
                15,
                32,
              ],
            },
            "spaced": true,
            "type": Symbol(Continuation),
            "value": "Line continuation",
          },
          Object {
            "line": 2,
            "ranges": Object {
              "directLineContinuationOperator": Array [
                33,
                34,
              ],
              "line": Array [
                33,
                55,
              ],
              "value": Array [
                35,
                55,
              ],
            },
            "spaced": false,
            "type": Symbol(Continuation),
            "value": "Newline continuation",
          },
        ],
        "key": "Field",
        "line": 0,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            5,
            6,
          ],
          "key": Array [
            0,
            5,
          ],
          "line": Array [
            0,
            12,
          ],
          "value": Array [
            7,
            12,
          ],
        },
        "type": Symbol(Field),
        "value": "Value",
      },
      Object {
        "entries": Array [
          Object {
            "continuations": Array [
              Object {
                "line": 6,
                "ranges": Object {
                  "line": Array [
                    81,
                    100,
                  ],
                  "spacedLineContinuationOperator": Array [
                    81,
                    82,
                  ],
                  "value": Array [
                    83,
                    100,
                  ],
                },
                "spaced": true,
                "type": Symbol(Continuation),
                "value": "Line continuation",
              },
              Object {
                "line": 7,
                "ranges": Object {
                  "directLineContinuationOperator": Array [
                    101,
                    102,
                  ],
                  "line": Array [
                    101,
                    123,
                  ],
                  "value": Array [
                    103,
                    123,
                  ],
                },
                "spaced": false,
                "type": Symbol(Continuation),
                "value": "Newline continuation",
              },
            ],
            "key": "Empty",
            "line": 5,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                73,
                74,
              ],
              "key": Array [
                67,
                72,
              ],
              "line": Array [
                67,
                80,
              ],
              "value": Array [
                75,
                80,
              ],
            },
            "type": Symbol(Attribute),
            "value": "Value",
          },
        ],
        "key": "Fieldset",
        "line": 4,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            65,
            66,
          ],
          "key": Array [
            57,
            65,
          ],
          "line": Array [
            57,
            66,
          ],
        },
        "type": Symbol(Fieldset),
      },
      Object {
        "items": Array [
          Object {
            "continuations": Array [
              Object {
                "line": 11,
                "ranges": Object {
                  "line": Array [
                    139,
                    158,
                  ],
                  "spacedLineContinuationOperator": Array [
                    139,
                    140,
                  ],
                  "value": Array [
                    141,
                    158,
                  ],
                },
                "spaced": true,
                "type": Symbol(Continuation),
                "value": "Line continuation",
              },
              Object {
                "line": 12,
                "ranges": Object {
                  "directLineContinuationOperator": Array [
                    159,
                    160,
                  ],
                  "line": Array [
                    159,
                    181,
                  ],
                  "value": Array [
                    161,
                    181,
                  ],
                },
                "spaced": false,
                "type": Symbol(Continuation),
                "value": "Newline continuation",
              },
            ],
            "line": 10,
            "parent": [Circular],
            "ranges": Object {
              "itemOperator": Array [
                131,
                132,
              ],
              "line": Array [
                131,
                138,
              ],
              "value": Array [
                133,
                138,
              ],
            },
            "type": Symbol(Item),
            "value": "Value",
          },
        ],
        "key": "List",
        "line": 9,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            129,
            130,
          ],
          "key": Array [
            125,
            129,
          ],
          "line": Array [
            125,
            130,
          ],
        },
        "type": Symbol(List),
      },
      Object {
        "continuations": Array [
          Object {
            "line": 15,
            "ranges": Object {
              "line": Array [
                196,
                215,
              ],
              "spacedLineContinuationOperator": Array [
                196,
                197,
              ],
              "value": Array [
                198,
                215,
              ],
            },
            "spaced": true,
            "type": Symbol(Continuation),
            "value": "Line continuation",
          },
          Object {
            "line": 16,
            "ranges": Object {
              "directLineContinuationOperator": Array [
                216,
                217,
              ],
              "line": Array [
                216,
                238,
              ],
              "value": Array [
                218,
                238,
              ],
            },
            "spaced": false,
            "type": Symbol(Continuation),
            "value": "Newline continuation",
          },
        ],
        "key": "Empty field",
        "line": 14,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            194,
            195,
          ],
          "key": Array [
            183,
            194,
          ],
          "line": Array [
            183,
            195,
          ],
        },
        "type": Symbol(Field),
      },
      Object {
        "entries": Array [
          Object {
            "continuations": Array [
              Object {
                "line": 20,
                "ranges": Object {
                  "line": Array [
                    281,
                    300,
                  ],
                  "spacedLineContinuationOperator": Array [
                    281,
                    282,
                  ],
                  "value": Array [
                    283,
                    300,
                  ],
                },
                "spaced": true,
                "type": Symbol(Continuation),
                "value": "Line continuation",
              },
              Object {
                "line": 21,
                "ranges": Object {
                  "directLineContinuationOperator": Array [
                    301,
                    302,
                  ],
                  "line": Array [
                    301,
                    323,
                  ],
                  "value": Array [
                    303,
                    323,
                  ],
                },
                "spaced": false,
                "type": Symbol(Continuation),
                "value": "Newline continuation",
              },
            ],
            "key": "Empty entry",
            "line": 19,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                279,
                280,
              ],
              "key": Array [
                267,
                278,
              ],
              "line": Array [
                267,
                280,
              ],
            },
            "type": Symbol(Attribute),
            "value": null,
          },
        ],
        "key": "Fieldset with empty entry",
        "line": 18,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            265,
            266,
          ],
          "key": Array [
            240,
            265,
          ],
          "line": Array [
            240,
            266,
          ],
        },
        "type": Symbol(Fieldset),
      },
      Object {
        "items": Array [
          Object {
            "continuations": Array [
              Object {
                "line": 25,
                "ranges": Object {
                  "line": Array [
                    349,
                    368,
                  ],
                  "spacedLineContinuationOperator": Array [
                    349,
                    350,
                  ],
                  "value": Array [
                    351,
                    368,
                  ],
                },
                "spaced": true,
                "type": Symbol(Continuation),
                "value": "Line continuation",
              },
              Object {
                "line": 26,
                "ranges": Object {
                  "directLineContinuationOperator": Array [
                    369,
                    370,
                  ],
                  "line": Array [
                    369,
                    391,
                  ],
                  "value": Array [
                    371,
                    391,
                  ],
                },
                "spaced": false,
                "type": Symbol(Continuation),
                "value": "Newline continuation",
              },
            ],
            "line": 24,
            "parent": [Circular],
            "ranges": Object {
              "itemOperator": Array [
                347,
                348,
              ],
              "line": Array [
                347,
                348,
              ],
            },
            "type": Symbol(Item),
            "value": null,
          },
        ],
        "key": "List with empty item",
        "line": 23,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            345,
            346,
          ],
          "key": Array [
            325,
            345,
          ],
          "line": Array [
            325,
            346,
          ],
        },
        "type": Symbol(List),
      },
    ],
    "type": Symbol(Document),
  },
  "_lineCount": 27,
  "_meta": Array [],
}
`;
